drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -0,0 +1,920 @@
+	const usb_device_controller_interface_struct_t *mcux_if = config->mcux_if;
+	struct udc_mcux_data *priv = udc_get_private(dev);
+	usb_status_t status = kStatus_USB_Success;
+	uint8_t *data;
+	uint32_t len;
+	unsigned int key;
+	usb_device_endpoint_status_struct_t ep_status;
+
+	ep_status.endpointAddress = cfg->addr;
+	usb_mcux_control(dev, kUSB_DeviceControlGetEndpointStatus, &ep_status);
+	if (ep_status.endpointStatus == kUSB_DeviceEndpointStateStalled) {
+		return -EACCES; /* stalled */
+	}
+
+	key = irq_lock();
+	if (!udc_ep_is_busy(dev, cfg->addr)) {
+		udc_ep_set_busy(dev, cfg->addr, true);
+		irq_unlock(key);
+
+		if (USB_EP_DIR_IS_OUT(cfg->addr)) {
+			len = net_buf_tailroom(buf);
+#if defined(CONFIG_NOCACHE_MEMORY)
+			data = (len == 0 ? NULL : udc_mcux_nocache_alloc(len));
+#else
+			data = net_buf_tail(buf);
+#endif
+			status = mcux_if->deviceRecv(priv->mcux_device.controllerHandle,
+					cfg->addr, data, len);
+		} else {
+			len = buf->len;
+#if defined(CONFIG_NOCACHE_MEMORY)
+			data = (len == 0 ? NULL : udc_mcux_nocache_alloc(len));
+			memcpy(data, buf->data, len);
+#else
+			data = buf->data;
+#endif
+			status = mcux_if->deviceSend(priv->mcux_device.controllerHandle,
+					cfg->addr, data, len);
+		}
+
+		key = irq_lock();
+		if (status != kStatus_USB_Success) {
+			udc_ep_set_busy(dev, cfg->addr, false);
+		}
+		irq_unlock(key);
+	} else {
+		irq_unlock(key);
+		return -EBUSY;
+	}
+
+	return (status == kStatus_USB_Success ? 0 : -EIO);
+}
+
+/* return success if the ep is busy or stalled. */
+static int usb_mcux_ep_try_feed(const struct device *dev,
+			struct udc_ep_config *const cfg)
+{
+	struct net_buf *feed_buf;
+
+	feed_buf = udc_buf_peek(dev, cfg->addr);
+	if (feed_buf) {
+		int ret = usb_mcux_ep_feed(dev, cfg, feed_buf);
+
+		return ((ret == -EBUSY || ret == -EACCES || ret == 0) ? 0 : -EIO);
+	}
+
+	return 0;
+}
+
+/*
+ * Allocate buffer and initiate a new control OUT transfer.
+ */
+static int usb_mcux_ctrl_feed_dout(const struct device *dev,
+				   const size_t length)
+{
+	struct net_buf *buf;
+	struct udc_ep_config *cfg = udc_get_ep_cfg(dev, USB_CONTROL_EP_OUT);
+	int ret;
+
+	buf = udc_ctrl_alloc(dev, USB_CONTROL_EP_OUT, length);
+	if (buf == NULL) {
+		return -ENOMEM;
+	}
+
+	net_buf_put(&cfg->fifo, buf);
+
+	ret = usb_mcux_ep_feed(dev, cfg, buf);
+
+	if (ret) {
+		net_buf_unref(buf);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int udc_mcux_handler_setup(const struct device *dev, struct usb_setup_packet *setup)
+{
+	int err;
+	struct net_buf *buf;
+
+	LOG_DBG("setup packet");
+	buf = udc_ctrl_alloc(dev, USB_CONTROL_EP_OUT,
+			sizeof(struct usb_setup_packet));
+	if (buf == NULL) {
+		LOG_ERR("Failed to allocate for setup");
+		return -EIO;
+	}
+
+	udc_ep_buf_set_setup(buf);
+	memcpy(buf->data, setup, 8);
+	net_buf_add(buf, 8);
+
+	if ((setup->RequestType.type == USB_REQTYPE_TYPE_STANDARD) &&
+		(!setup->wLength) &&
+		(!(((setup->bmRequestType) >> 7) & 0x01U)) &&
```

Comment:
Please write it consistently, if you start with `setup->RequestType.type == USB_REQTYPE_TYPE_STANDARD`, then 

```c
	if (setup->RequestType.type == USB_REQTYPE_TYPE_STANDARD &&
	    setup->RequestType.direction == USB_REQTYPE_DIR_TO_DEVICE &&
	    setup->bRequest == USB_SREQ_SET_ADDRESS &&
	    setup->wLength == 0) {
```

Reply:
Updated.